pyscf/pbc/df/rsdf.py

```diff
@@ -0,0 +1,703 @@
+                 len(np.unique(scaled_k[:,1])),
+                 len(np.unique(scaled_k[:,2])))
+    else:
+        kmesh = None
+    return kmesh
+
+
+def weighted_coulG(cell, omega, kpt=np.zeros(3), exx=False, mesh=None):
+    if cell.omega != 0:
+        raise RuntimeError('RSGDF cannot be used '
+                           'to evaluate the long-range HF exchange in RSH '
+                           'functional')
+    Gv, Gvbase, kws = cell.get_Gv_weights(mesh)
+    if abs(omega) < 1.e-10:
+        omega_ = None
+    else:
+        omega_ = omega
+    coulG = pbctools.get_coulG(cell, kpt, False, None, mesh, Gv,
+                               omega=omega_)
+    coulG *= kws
+    return coulG
+def get_aux_chg(auxcell):
+    r""" Compute charge of the auxiliary basis, \int_Omega dr chi_P(r)
+
+    Returns:
+        The function returns a 1d numpy array of size auxcell.nao_nr().
+    """
+    def get_nd(l):
+        if auxcell.cart:
+            return (l+1) * (l+2) // 2
+        else:
+            return 2 * l + 1
+
+    naux = auxcell.nao_nr()
+    qs = np.zeros(naux)
+    shift = 0
+    half_sph_norm = np.sqrt(4*np.pi)
+    for ib in range(auxcell.nbas):
+        l = auxcell.bas_angular(ib)
+        if l == 0:
+            npm = auxcell.bas_nprim(ib)
+            nc = auxcell.bas_nctr(ib)
+            es = auxcell.bas_exp(ib)
+            ptr = auxcell._bas[ib,mol_gto.PTR_COEFF]
+            cs = auxcell._env[ptr:ptr+npm*nc].reshape(nc,npm).T
+            norms = mol_gto.gaussian_int(l+2, es)
+            q = np.einsum("i,ij->j",norms,cs)[0] * half_sph_norm
+        else:   # higher angular momentum AOs carry no charge
+            q = 0.
+        nd = get_nd(l)
+        qs[shift:shift+nd] = q
+        shift += nd
+
+    return qs
+
+
+# kpti == kptj: s2 symmetry
+# kpti == kptj == 0 (gamma point): real
+def _make_j3c(mydf, cell, auxcell, kptij_lst, cderi_file):
+    t1 = (logger.process_clock(), logger.perf_counter())
+    log = logger.Logger(mydf.stdout, mydf.verbose)
+    max_memory = max(2000, mydf.max_memory-lib.current_memory()[0])
+
+    omega = abs(mydf.omega)
+
+    if mydf.use_bvk:
+        bvk_kmesh = kpts_to_kmesh(cell, mydf.kpts)
+        if bvk_kmesh is None:
+            log.debug("Non-Gamma-inclusive kmesh is found. bvk kmesh is not used.")
+        else:
+            log.debug("Using bvk kmesh= [%d %d %d]", *bvk_kmesh)
+    else:
+        bvk_kmesh = None
+
+    # The ideal way to hold the temporary integrals is to store them in the
+    # cderi_file and overwrite them inplace in the second pass.  The current
+    # HDF5 library does not have an efficient way to manage free space in
+    # overwriting.  It often leads to the cderi_file ~2 times larger than the
+    # necessary size.  For now, dumping the DF integral intermediates to a
+    # separated temporary file can avoid this issue.  The DF intermediates may
+    # be terribly huge. The temporary file should be placed in the same disk
```

Comment:
Why should it be placed on the same disk?

Reply:
these lines were copied from GDF